ci/do_ci.sh

```diff
@@ -232,6 +232,9 @@ elif [[ "$CI_TARGET" == "bazel.compile_time_options" ]]; then
   # Right now, none of the available compile-time options conflict with each other. If this
   # changes, this build type may need to be broken up.
   # TODO(mpwarres): remove quiche=enabled once QUICHE is built by default.
+  # TODO(dio): To resolve the conflicted zlib=ng vs. zlib=ng_with_optimizations. However, it should
+  # be safe to only build zlib=ng_with_optimizations for now, since these two compile options use
+  # the same source and most likely to build with "ng_with_optimizations" is what people want.
```

Comment:
Sorry in looking at this again, do we really need to have two versions here? Can we just key off of `-c opt` or not? cc @lizan 

/wait

Reply:
The `zlib=ng` uses `zlib-ng` but optimizations turned off, provides complete compatibility with upstream `zlib`. This is introduced since there are possibilities optimizations break tests, so this switch provides an easy switch to check if that is the case (e.g. when we sync upstream `zlib-ng` changes).

Reply:
Oh, sorry. I think I misread the comment. So we should honor the `-c opt` for switching on optimizations?

Reply:
You can always use the CMake entries for ng_with_optimizations. Bazel will still add debug symbols to debug builds so you won't loose them. Or you can also use `debug_cache_entries` to add entries to debug build if you want to disable optimization there.

Reply:
> Oh, sorry. I think I misread the comment. So we should honor the -c opt for switching on optimizations?

That's what I'm suggesting. Just switch the internal build based on whether `-c opt` or not and just have a single compile flag for using this library?

Reply:
We've being running the optimizations for a while and they seem to be safe, so I am OK with gating them with `-c opt`. However, I am not sure if this is universally safe for everyone....

Reply:
So probably, the following fits the bill? 

The complete compatible `zlib-ng` flags:

```
build --define zlib=ng
```

For turning on the zlib-ng's `WITH_OPTIM` flag.

```
build --define zlib=ng -c opt
```

Provided by:

```bazel
config_setting(
    name = "zlib_ng",
    constraint_values = [
        "@bazel_tools//platforms:linux",
    ],
    values = {"define": "zlib=ng"},
)

config_setting(
    name = "zlib_ng_with_optimizations",
    constraint_values = [
        "@bazel_tools//platforms:linux",
    ],
    values = {
        "define": "zlib=ng",
        "compilation_mode": "opt",
    },
)
```

Reply:
Sorry what does WITH_OPTIM do? I just assumed that is optimized code? Does it do something else?

Reply:
It turns on some arch-specifics optimizations: https://github.com/zlib-ng/zlib-ng/blob/1d2504ddc4894786fdf61d41a9bfa435cd8b1935/CMakeLists.txt#L646-L756.

Reply:
OK I see, it's not about compile type. My vote would be to just get rid of the two variants. I would always turn on the optimizations and if someone complains we can add back the other option? Then opt/dbg/etc. should do the right thing wrt to  compile flags?

Reply:
OK. So we only have `--define zlib=ng` now, with selected `zlib-ng` build options to turn `WITH_OPTIM` on.